azure-iot-device/samples/advanced-hub-scenarios/register_x509.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+
+# This is for illustration purposes only. The sample will not work currently.
+
+import os
+import logging
+import asyncio
+from azure.iot.device import X509SecurityClient, X509
+from azure.iot.device.aio import X509ProvisioningDeviceClient
+
+
+logging.basicConfig(level=logging.INFO)
+
+provisioning_host = os.getenv("PROVISIONING_HOST")
+id_scope = os.getenv("PROVISIONING_IDSCOPE")
+registration_id = os.getenv("DPS_X509_REGISTRATION_ID")
+
+
+async def main():
+    async def register_device():
+        x509_cert = X509(
```

Comment:
This is now 3 step instantiation.

Not a huge deal right now, this could probably wait to be addressed until a broad provisioning client API revision.

Reply:
yes. the only way this can be 1 step is to male the client creation take 6 params...it can be done..but we can discuss this further

---
In reply to: [291751154](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291751154) [](ancestors = 291751154)

Reply:
i would like to revisit this after client changes. Coz these 2 changes are going to affect each other.

---
In reply to: [291789509](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291789509) [](ancestors = 291789509,291751154)